### PR TITLE
Added requirements.txt file, Added dockerfile gpu cuda 10.2 [WIP]

### PR DESCRIPTION
**Pull request by ghost**
- support for cpu and gpu (cuda 10.2)
- added requirements.txt file

### CODE REVIEW

**Comment by ghost** [comment]
For now, I have the following error when I run:

```sh
$ docker run -ti -p 4000:4000 semantic-sh:latest --port=4000
Exception: To use fasttext embeddings, model_path must be given.
```

Is there a way to load the model while starting the server ?

Cheers,
X

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+version: '3.8'
+services:
+ 
+   semantic-sh:
```
Comment (KeremZaman) [inline docker-compose.yml:5]: Providing commandline parameters to server.py by defining environment variables in docker-compose will give more configurable docker usage.  I'm not sure the code I write below will work properly, please test before committing.
```suggestion
   semantic-sh:
     environment:
    - model_type
    - model_path
    - key_size
    - dim
    - host
    - port
```

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+     image: semantic-sh:latest
+     container_name: semantic-sh
+     build:
+       context: .
+       dockerfile: Dockerfile
+     ports:
+     - 4000:4000
+     volumes:
+     - models-data:/opt/data
+     command: ["--port", "4000", "--model-type", "bert-base-multilingual-cased", "--model-path", "/opt/data"]
```
Comment (KeremZaman) [inline docker-compose.yml]: Use defined environment values here. 
```suggestion
     command: ["--host", ${host}, "--port", ${port}, "--model-type", ${model_type}, "--model-path", ${model_path}, "--dim", ${dim}, "--key-size", ${key_size}]
```

**Review by KeremZaman — CHANGES_REQUESTED** [review]
I made some comments for this PR, please make sure that you will test properly after changes.

File: docker-compose.yml (hunk 1 above)
Comment (ghost) [reply docker-compose.yml:5]: Thanks for the comment, shall I create an .env file ? 
rather than using hard coded values in the docker-compose.yml file.

Comment (x0rzkov) [reply docker-compose.yml:5]: What are your recommended/default values for ?  
```txt
--key-size KEY_SIZE   Hash length in bits
--dim DIM             Dimension of text representations according to chosen model type
```

Comment (KeremZaman) [reply docker-compose.yml:5]: Yes, it'd probably better.

Comment (ghost) [reply docker-compose.yml:5]: That's what I did, in my next push :-)

**Pull request merged**